plugins/source-kafka/pkg/client/client.go

```diff
@@ -15,9 +15,139 @@
 package client
 
 import (
+	sourceclient "github.com/maximilien/kn-source-pkg/pkg/client"
+	sourcetypes "github.com/maximilien/kn-source-pkg/pkg/types"
+	"knative.dev/client-contrib/plugins/source-kafka/pkg/types"
+	knerrors "knative.dev/client/pkg/errors"
+	v1alpha1 "knative.dev/eventing-contrib/kafka/source/pkg/apis/sources/v1alpha1"
 	clientv1alpha1 "knative.dev/eventing-contrib/kafka/source/pkg/client/clientset/versioned/typed/sources/v1alpha1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
 )
 
 type kafkaSourceClient struct {
-	client clientv1alpha1.SourcesV1alpha1Interface
+	kafkaSourceParams *types.KafkaSourceParams
+	client            clientv1alpha1.SourcesV1alpha1Interface
+	namespace         string
+	knSourceClient    sourcetypes.KnSourceClient
+}
+
+func NewKafkaSourceClient(kafkaParams *types.KafkaSourceParams, c clientv1alpha1.SourcesV1alpha1Interface, ns string) types.KafkaSourceClient {
+	if c == nil {
+		c, _ = kafkaParams.NewSourcesClient()
+	}
+	return &kafkaSourceClient{
+		kafkaSourceParams: kafkaParams,
+		client:            c,
+		namespace:         ns,
+		knSourceClient:    sourceclient.NewKnSourceClient(kafkaParams.KnSourceParams, ns),
+	}
+}
+
+// RestConfig the REST cconfig
+func (c *kafkaSourceClient) RestConfig() *rest.Config {
+	return c.knSourceClient.RestConfig()
+}
+
+func (c *kafkaSourceClient) KnSourceParams() *sourcetypes.KnSourceParams {
```

Comment:
Add comment otherwise /lint will complain. See how I moved many of these into a `boillerplate.go` in the `kn-source-github` file which you might be able to reuse as well as test.

Reply:
fixed.